src/datatypes/sipath.c

```diff
@@ -59,8 +40,7 @@ SIValue SIPath_ToList(SIValue p) {
 }
 
 SIValue SIPath_Relationships(SIValue p) {
```

Comment:
if `SIPath_Relationships` returned value is consider immutable from the perspective of the caller(s) then we can avoid creating a clone here and return a direct reference to the under-line array.
The same goes for `SIPath_Nodes`

Reply:
As far as I see this, we can't pass a direct reference.
The only usage for those functions is where the user asks for relationships or nodes of the path with the functions `relationships(p)` and `nodes(p), respectfully. 
Now consider the following query `MATCH p=(a)-[*]->(b) with relationships(p) as x return size(x)`. If we pass a direct reference the edges array of `p`, after the projection of the WITH clause, the path object is freed, due to the record free, and x is now a pointer to invalid memory.

Reply:
I see, thanks!